src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -640,60 +640,80 @@ def _set_linux_user(
-                                                                _ACCESS_EXT_HANDLER_NAME, ext)
+    poller = client.virtual_machine_extensions.create_or_update(resource_group_name,
+                                                                vm_instance.name,
+                                                                instance_name, ext)
     return ExtensionUpdateLongRunningOperation('resetting admin', 'done')(poller)
 
 
-def _update_linux_access_extension(resource_group_name, vm_name, protected_settings):
-    vm = get_vm(resource_group_name, vm_name, 'instanceView')
+def _update_linux_access_extension(vm_instance, resource_group_name, protected_settings):
     client = _compute_client_factory()
 
     from azure.mgmt.compute.models import VirtualMachineExtension
+    # pylint: disable=no-member
+    instance_name = _get_extension_instance_name(vm_instance.instance_view,
+                                                 extension_mappings[_LINUX_ACCESS_EXT]['publisher'],
+                                                 _LINUX_ACCESS_EXT,
+                                                 _ACCESS_EXT_HANDLER_NAME)
 
-    extension_name = _LINUX_ACCESS_EXT
     publisher, version, auto_upgrade = _get_access_extension_upgrade_info(
-        vm.resources, extension_name)
+        vm_instance.resources, _LINUX_ACCESS_EXT)
 
-    ext = VirtualMachineExtension(vm.location,  # pylint: disable=no-member
+    ext = VirtualMachineExtension(vm_instance.location,  # pylint: disable=no-member
                                   publisher=publisher,
-                                  virtual_machine_extension_type=extension_name,
+                                  virtual_machine_extension_type=_LINUX_ACCESS_EXT,
                                   protected_settings=protected_settings,
                                   type_handler_version=version,
                                   settings={},
                                   auto_upgrade_minor_version=auto_upgrade)
 
-    poller = client.virtual_machine_extensions.create_or_update(resource_group_name, vm_name,
-                                                                _ACCESS_EXT_HANDLER_NAME, ext)
+    poller = client.virtual_machine_extensions.create_or_update(resource_group_name,
+                                                                vm_instance.name,
+                                                                instance_name, ext)
     return poller
 
 
+def _get_extension_instance_name(instance_view, publisher, extension_type_name,
+                                 suggested_name=None):
+    extension_instance_name = suggested_name or extension_type_name
+    full_type_name = '.'.join([publisher, extension_type_name])
+    if instance_view.extensions:
+        ext = next((x for x in instance_view.extensions
+                    if x.type.lower() == full_type_name.lower()))
+        if ext:
+            extension_instance_name = ext.name
```

Comment:
If there is no existing extension with that type, this routine will use the type as the default name; otherwise it will be the name of the existing one. For background, you can have only one extension with that type (say, LinuxAccess, LinuxDiagnostic, or AzureDiskEncryptionForLinux), so I decided not to expose the name as an arg.